### PR TITLE
Move AadAuthorityCache to common core

### PR DESCRIPTION
**Pull request by jasoncoolmax**
Some meta data will be stored when we do authority validation for AAD authorities.
Such a cache is now moved to common core, such that it can be shared by ADAL and MSAL.

Note that MSAL is now not filling anything to such a cache, but it will when we unify the authority validation later.

### CODE REVIEW

File: IdentityCore/src/validation/MSIDAuthority.m [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//
+//------------------------------------------------------------------------------
+
+#import "MSIDAuthority.h"
+
+@implementation MSIDAuthority
+
++ (BOOL)isADFSInstance:(NSString *)endpoint
+{
+    if([NSString msidIsStringNilOrBlank:endpoint]){
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDAuthority.m]: nit: whitespaces

File: IdentityCore/src/validation/MSIDAuthority.m [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return NO;
+    }
+    
+    return[[self class] isADFSInstanceURL: [NSURL URLWithString:endpoint.lowercaseString]];
+}
+
++ (BOOL)isADFSInstanceURL:(NSURL *)endpointUrl
+{
+    
+    NSArray* paths = endpointUrl.pathComponents;
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDAuthority.m]: very nit: can we keep star usage consistent for new code (NSArray *paths)

File: IdentityCore/src/validation/MSIDAuthority.m (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/validation/MSIDAuthority.m]: I actually meant to add whitespaces and not remove them... Sorry for not clarifying :), e.g.

`if ([NSString msidIsStringNilOrBlank:endpoint]){`

or

`return [[self class] isADFSInstanceURL:[NSURL URLWithString:endpoint.lowercaseString]];`

Comment (jasoncoolmax) [reply IdentityCore/src/validation/MSIDAuthority.m]: Oh got it :)

**Pull request merged**